overlord/ifacestate/ifacestate_test.go

```diff
@@ -10696,3 +10696,134 @@ func (s *interfaceManagerSuite) TestOnSnapLinkageChanged(c *C) {
 		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{&info.SideInfo}),
 	})
 }
+
+func (s *interfaceManagerSuite) TestSetupProfilesAffectedSnapRegenUsesMostRecentRevision(c *C) {
+	// this test checks a bug fix where a consumer and producer snap (content interface)
+	// were refreshed together and sometimes the producing snap lost access to its own
+	// file. The issue was that, if the slot (producer) was refreshed first, then,
+	// when the plugging snap was refreshed it would regenerate the slots profile
+	// but with the previous revision, not the revision that the current change was
+	// updating to.
+	s.mockIfaces(&ifacetest.TestInterface{InterfaceName: "test"})
+
+	const componentYaml = `
+component: producer2+comp1
```

Comment:
What's with the 2 there?

Reply:
It's just the name of the snap `producer2`